store/sqlstore/post_store.go

```diff
@@ -2928,6 +2929,22 @@ func (s *SqlPostStore) updateThreadsFromPosts(transaction *sqlxTxWrapper, posts
 		threadByRoot[thread.PostId] = thread
 	}
 
+	// let's grab all different teamIds from the posts
+	teamIdByChannelId := map[string]string{}
+	for _, posts := range postsByRoot {
+		channelId := posts[0].ChannelId
+		if _, ok := teamIdByChannelId[channelId]; !ok {
+			// get teamId for channel
+			var teamId string
+			err = transaction.Get(&teamId, "SELECT COALESCE(Channels.TeamId, '') FROM Channels WHERE Channels.Id=?", channelId)
+			if err != nil {
+				return err
+			}
+
+			teamIdByChannelId[channelId] = teamId
+		}
+	}
```

Comment:
It might be more efficient to only do this for posts for which threads don't exist. So within the conditional on line 2949. 

We don't need this look up for posts that already have a thread entry (therefore they already have the team id).